### DISCUSSION TITLE
Original data like models have and returning it merged

### DISCUSSION THREAD

**Discussion by wijzijnweb** [spatie/laravel-data#606] [Ideas]
Hi there,

Not sure if this is a good idea or not.
But I stumbled upon a situation where I did add some dynamic attributes to a model.

As soon as I transformed it to a Data class those attributes were gone because it isn't a property on the Data class.
I don't really want to add it on the Data class since it's data that isn't used everywhere just a combined label (string) from a grouped model. 

I came up with this in my BaseData class:

```
public static function from(...$payloads): static
{
    $data = parent::from(...$payloads);
    $data->original = $payloads[0] ?? [];
    return $data;
}

public function __get(string $name)
{
    return $this->{$name} ?? $this->original[$name] ?? null;
}

public function toArray(): array
{
    $array = parent::toArray();

    $dotted = Arr::dot($array);
    $originalDotted = Arr::dot($array['original']);

    foreach($originalDotted as $key => $value) {
        $realKey = str_replace('original.', '', $key);
        if(!isset($dotted[$realKey])) {
            $dotted[$realKey] = $value;
        }
    }

    $array = Arr::undot($dotted);
    unset($array['original']);
    return $array;
}
``` 

Wondering if any of you think this is a good idea or maybe keeping the original data aside?

**Comment by rubenvanassche** [comment]
It is an interesting solution, I myself would keep this somewhere apart if it is required. Or just set the properties. But if it works that's great!
